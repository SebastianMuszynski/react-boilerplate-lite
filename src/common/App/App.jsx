import React, { Component } from "react";
import { createPortal } from "react-dom";

const portals = [];

function createSRRPortal(children, selector) {
  if (typeof window === "undefined") {
    portals.push([selector, children]);
    return null;
  }
  return createPortal(children, document.querySelector(selector));
}

export function flushSSRPortals() {
  const copy = [...portals];
  portals.length = 0;
  return copy;
}

const Head = ({ children }) => createSRRPortal(children, "head");

const Article = ({ title, description, children }) => (
  <article>
    <Head>
      <title>{title}</title>
      <meta name="description" content={description} />
    </Head>
    <h1>{title}</h1>
    {children}
  </article>
);

class App extends Component {
  state = {
    article: 0,
  };

  increment = () => {
    this.setState(({ article }) => ({ article: article + 1 }));
  };

  render() {
    if (this.state.article % 2) {
      return (
        <Article title="Hello, Portals!" description="Awesome">
          <p>Lorem ipsum!</p>
          <button onClick={this.increment}>Button</button>
        </Article>
      );
    }

    return (
      <Article title="Hello, World!" description="eh, ok">
        <p>Lorem ipsum?</p>
        <button onClick={this.increment}>Button</button>
      </Article>
    );
  }
}

export default App;
