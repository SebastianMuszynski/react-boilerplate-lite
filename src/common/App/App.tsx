import * as React from "react";
const cls = require("./App.css");

interface AppStateType {
  text: string;
}

class App extends React.Component<{}, AppStateType> {
  state = {
    text: "Hello World!",
  };

  render() {
    return (
      <div>
        <h1 className={cls.title}>{this.state.text}</h1>
      </div>
    );
  }
}

export default App;
