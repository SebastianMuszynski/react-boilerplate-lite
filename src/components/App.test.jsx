import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import App from "./App";

test("Contains text from state", () => {
  const tree = shallow(<App />);

  tree.setState({ text: "Foo!" });

  expect(tree.find(".title").contains("Foo!")).toBe(true);

  expect(toJson(tree)).toMatchSnapshot();
});
