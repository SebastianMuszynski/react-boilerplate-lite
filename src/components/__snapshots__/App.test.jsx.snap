// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contains text from state 1`] = `
<div>
  <h1
    className="title"
  >
    Foo!
  </h1>
</div>
`;
